### PR TITLE
Add a dart dev_server for each web app.

### PR DESCRIPTION
**Pull request by matanlurey**
/cc @kevmoo @jakemac53 @nshahan

### CODE REVIEW

**Comment by natebosch** [comment]
We also need to tag rules_dart and update the generated WORKSPACE for this to work

**Comment by nshahan** [comment]
The dev_server doesn't work either. I believe it is expecting a dependency on a ddc_bundle.

File: lib/src/bazelify/build.dart [hunk 1]
```diff
@@ -282,14 +285,21 @@ class DartWebApplication implements DartBuildRule {
...
-          ' + [\n' +
+      buffer += ' + [\n' +
           includeLibraries
               .map/*<String>*/((l) => '        ":${l.name}",\n')
               .join() +
           '    ],\n)';
     }
+    buffer += '\ndev_server(\n'
+        '    name = "${name}_serve",\n'
+        '    deps = _PUB_DEPS + [\n' +
```
Comment (nshahan) [inline lib/src/bazelify/build.dart]: Can we leave out _PUB_DEPS if we are also depending on the library that already has _PUB_DEPS?

File: test/goldens/build_file_web_application [hunk 2]
```diff
@@ -28,3 +28,10 @@ dart_web_application(
         ":silly_monkey",
     ],
 )
+dev_server(
+    name = "main_web_serve",
+    deps = _PUB_DEPS + [
+        ":silly_monkey",
+    ],
+    data = glob(["web/**"]),
```
Comment (nshahan) [inline test/goldens/build_file_web_application]: Shouldn't need data. dev_server should have everything it needs from deps.

File: test/goldens/build_file_web_application [hunk 3]
```diff
@@ -28,3 +28,10 @@ dart_web_application(
         ":silly_monkey",
     ],
 )
+dev_server(
+    name = "main_web_serve",
+    deps = _PUB_DEPS + [
```
Comment (nshahan) [inline test/goldens/build_file_web_application]: Can leave out _PUB_DEPS here too?

**Comment by matanlurey** [comment]
Closing and re-doing from the new fork.

File: test/goldens/build_file_web_application (hunk 3 above)
Comment (matanlurey) [inline test/goldens/build_file_web_application]: Acknowledged. Let's do in a follow-up if this isn't breaking.

File: test/goldens/build_file_web_application (hunk 2 above)
Comment (matanlurey) [inline test/goldens/build_file_web_application]: Acknowledged. Let's do in a follow-up if this isn't breaking.

**Pull request closed without merging**